### PR TITLE
Feat/create the network class for gui

### PR DESCRIPTION
**Pull request by mathieurobert1**
I create the Network class to handle messages from the Zappy server and the Error class.

More details : #4 

Here the utilization of the class:
image

### CODE REVIEW

**Review by Njord201 — DISMISSED** [review]

File: gui/include/Error/AError.hpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        /**
+         * @brief Destructor.
+         */
+        ~AError() override = default;
+
+        /**
+         * @brief Returns the error message.
+         * @return A pointer to a constant character string representing the error message.
+         */
+        const char *what() const noexcept override { return _message.c_str(); }
```
Comment (Thomaltarix) [inline gui/include/Error/AError.hpp:40]: You should put this method code into a cpp file

File: gui/include/Error/Error.hpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+         * @class ArgumentNumber
+         * @brief Error class for network errors.
+         */
+        class Network : public Error {
+            public:
+                /**
+                 * @brief Constructor for Network.
+                 * @param message The error message.
+                 */
+                Network(std::string message) { _message = message; }
```
Comment (Thomaltarix) [inline gui/include/Error/Error.hpp:31]: Same here, you should put the method code into a cpp file

File: gui/include/Network/Network.hpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+         * @param message to send to the server
+         */
+        void sendMessageServer(const std::string& message);
+
+        int             _port;
+        std::string     _hostName;
+        int             _serverFd;
+        fd_set          _writeFd;
+        fd_set          _readFd;
+        bool            _isConnected;
```
Comment (Thomaltarix) [inline gui/include/Network/Network.hpp:112]: You can had comments for this attributes

**Review by Thomaltarix — CHANGES_REQUESTED** [review]
Some changes, but overall that's a great job

File: gui/include/Error/Error.hpp (hunk 2 above)
Comment (Marius-P1) [reply gui/include/Error/Error.hpp:31]: +1

File: gui/include/Network/Network.hpp (hunk 3 above)
Comment (Marius-P1) [reply gui/include/Network/Network.hpp:112]: +1

**Review by Marius-P1 — CHANGES_REQUESTED** [review]
Some small changes

File: gui/include/Error/AError.hpp [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+         * It contains a protected member `_message` to store the error message.
+         */
+        class AError;
+    };
+};
+
+class Gui::Errors::AError : public IError {
+
+    public:
+
```
[the remark this answers is not in the document]
Comment (Thomaltarix) [reply gui/include/Error/AError.hpp:30]: If i'm not making a mistake too, i've tried for another project and it didn't work.  For each class that will inherit from `AError` and have no constructors, you wont be able to create an Error with this class.

For exemple, You have a `TestError` class which inherits from `AError`, you wont be able to call this `throw TestError("sample of error message")`

File: gui/src/Network/Network.cpp [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+Gui::Network::Network(int port, const std::string& hostName)
+{
+    setPort(port);
+    setHostName(hostName);
+    _isConnected = false;
+}
+
+void Gui::Network::setPort(int port)
+{
+    if (port == 0)
```
[the remark this answers is not in the document]
Comment (mathieurobert1) [reply gui/src/Network/Network.cpp:22]: Oh yes it's a mistake

File: gui/include/Network/Network.hpp (hunk 3 above)
Comment (mathieurobert1) [reply gui/include/Network/Network.hpp:112]: Let's debate about that with all the team's members

File: gui/include/Error/AError.hpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (mathieurobert1) [reply gui/include/Error/AError.hpp:30]: It didn't work

File: gui/src/Network/Network.cpp [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+** Network
+*/
+
+#include "Network/Network.hpp"
+
+#include <unistd.h>
+#include <iostream>
+
+#define MAX_PORT 65535
+#define MIN_PORT 1
```
Comment (Thomaltarix) [inline gui/src/Network/Network.cpp:14]: It should probably go into the header file

File: gui/src/Network/Network.cpp [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+{
+    setPort(port);
+    setHostName(hostName);
+    _isConnected = false;
+}
+
+void Gui::Network::setPort(int port)
+{
+    if (port < MIN_PORT || port > MAX_PORT)
+        throw Errors::Network("Port should not be equal to 0.");
```
Comment (Thomaltarix) [inline gui/src/Network/Network.cpp:26]: The error message is not updated. It should be something like "Serveur port must take part between 1 and 65535".

**Review by Thomaltarix — CHANGES_REQUESTED** [review]
A little change to do, then I'm done

**Review by Thomaltarix — APPROVED** [review]
Great

**Review by Marius-P1 — APPROVED** [review]
👌

**Review by Njord201 — APPROVED** [review]

**Pull request merged**